core/scripts/common/helpers.go

```diff
@@ -1,20 +1,111 @@
 package common
 
 import (
+	"context"
+	"crypto/ecdsa"
+	"encoding/hex"
 	"flag"
 	"fmt"
+	"math/big"
+	"os"
+	"strconv"
+	"strings"
 
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/shopspring/decimal"
 )
 
-// PanicErr panic if error detected
+type Environment struct {
+	Owner   *bind.TransactOpts
+	Ec      *ethclient.Client
+	ChainID int64
+}
+
+// SetupEnv returns an Environment object populated from environment variables.
+// If overrideNonce is set to true, the nonce will be set to what is returned
+// by NonceAt (rather than the typical PendingNonceAt).
+func SetupEnv(overrideNonce bool) Environment {
+	ethURL, set := os.LookupEnv("ETH_URL")
+	if !set {
+		panic("need eth url")
+	}
+
+	chainIDEnv, set := os.LookupEnv("ETH_CHAIN_ID")
+	if !set {
+		panic("need chain ID")
+	}
+
+	accountKey, set := os.LookupEnv("ACCOUNT_KEY")
+	if !set {
+		panic("need account key")
+	}
+
+	ec, err := ethclient.Dial(ethURL)
+	PanicErr(err)
+
+	chainID, err := strconv.ParseInt(chainIDEnv, 10, 64)
+	PanicErr(err)
+
+	// Owner key. Make sure it has eth
+	b, err := hex.DecodeString(accountKey)
+	PanicErr(err)
+	d := new(big.Int).SetBytes(b)
+
+	pkX, pkY := crypto.S256().ScalarBaseMult(d.Bytes())
+	privateKey := ecdsa.PrivateKey{
+		PublicKey: ecdsa.PublicKey{
+			Curve: crypto.S256(),
+			X:     pkX,
+			Y:     pkY,
+		},
+		D: d,
+	}
+	owner, err := bind.NewKeyedTransactorWithChainID(&privateKey, big.NewInt(chainID))
+	PanicErr(err)
+	// Explicitly set gas price to ensure non-eip 1559
+	gp, err := ec.SuggestGasPrice(context.Background())
+	PanicErr(err)
+	owner.GasPrice = gp
+	gasLimit, set := os.LookupEnv("GAS_LIMIT")
+	if set {
+		parsedGasLimit, err := strconv.ParseUint(gasLimit, 10, 64)
+		if err != nil {
+			panic(fmt.Sprintf("Failure while parsing GAS_LIMIT: %s", gasLimit))
+		}
+		owner.GasLimit = parsedGasLimit
+	}
+
+	if overrideNonce {
+		block, err := ec.BlockNumber(context.Background())
+		PanicErr(err)
+
+		nonce, err := ec.NonceAt(context.Background(), owner.From, big.NewInt(int64(block)))
+		PanicErr(err)
+
+		owner.Nonce = big.NewInt(int64(nonce))
+		owner.GasPrice = gp.Mul(gp, big.NewInt(2))
+	}
```

Comment:
Put this logic behind a boolean so that the runner of the script can enable/disable it by simply changing the call to `helpers.SetupEnv(true)` if they want this feature.